File: VB/ChangeEditMask.Module.Win/Controllers/WinChangeMaskController.vb
Imports ChangeEditMask.Module.Controllers
Imports DevExpress.ExpressApp.Win.Editors
Imports DevExpress.XtraEditors

Namespace ChangeEditMask.Module.Win.Controllers

    Public Class WinChangeMaskController
        Inherits ChangeMaskControllerBase

        Protected Overrides Sub SetControlMaskSettings(ByVal propertyEditor As DevExpress.ExpressApp.Editors.PropertyEditor, ByVal mask As BusinessObjects.EditMask)
            If TypeOf propertyEditor Is StringPropertyEditor Then
                Dim textEdit As TextEdit = CType(propertyEditor, StringPropertyEditor).Control
                Select Case mask
                    Case BusinessObjects.EditMask.Date
                        textEdit.Properties.Mask.MaskType = DevExpress.XtraEditors.Mask.MaskType.DateTime
                        textEdit.Properties.Mask.EditMask = "d"
                    Case BusinessObjects.EditMask.Time
                        textEdit.Properties.Mask.MaskType = DevExpress.XtraEditors.Mask.MaskType.DateTime
                        textEdit.Properties.Mask.EditMask = "t"
                    Case BusinessObjects.EditMask.Numeric
                        textEdit.Properties.Mask.MaskType = DevExpress.XtraEditors.Mask.MaskType.Numeric
                        textEdit.Properties.Mask.EditMask = "d"
                    Case BusinessObjects.EditMask.String
                        textEdit.Properties.Mask.MaskType = DevExpress.XtraEditors.Mask.MaskType.None
                        textEdit.Properties.Mask.EditMask = ""
                End Select
            End If
        End Sub
    End Class
End Namespace

File: CS/XPO/DynamicMask/DynamicMask.Win/Controllers/WinChangeMaskController.cs
using DevExpress.ExpressApp.Editors;
using DevExpress.ExpressApp.Win.Editors;
using DevExpress.XtraEditors;
using DevExpress.XtraEditors.Mask;
using DynamicMask.Module.BusinessObjects;
using DynamicMask.Module.Controllers;

namespace DynamicMask.Module.Win.Controllers {
    public class WinChangeMaskController : ChangeMaskControllerBase {
        protected override void SetControlMaskSettings(PropertyEditor propertyEditor, EditMask mask) {
            if (propertyEditor is StringPropertyEditor) {
                TextEdit textEdit = ((StringPropertyEditor)propertyEditor).Control;
                switch (mask) {
                    case EditMask.Date:
                        textEdit.Properties.Mask.MaskType = MaskType.DateTime;
                        textEdit.Properties.Mask.EditMask = "d";
                        break;
                    case EditMask.Time:
                        textEdit.Properties.Mask.MaskType = MaskType.DateTime;
                        textEdit.Properties.Mask.EditMask = "t";
                        break;
                    case EditMask.Numeric:
                        textEdit.Properties.Mask.MaskType = MaskType.Numeric;
                        textEdit.Properties.Mask.EditMask = "d";
                        break;
                    case EditMask.String:
                        textEdit.Properties.Mask.MaskType = MaskType.None;
                        textEdit.Properties.Mask.EditMask = "";
                        break;
                }
            }
        }
    }
}
